addons/mail/tests/test_mail_activity_plan.py

```diff
@@ -0,0 +1,46 @@
+from odoo import Command
+from odoo.addons.mail.tests.test_mail_activity import ActivityScheduleCase
+from odoo.tests import tagged
+
+
+class ActivityPlanScheduleCase(ActivityScheduleCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.test_company = cls.env['res.company'].create({
+            'name': 'Test Company',
+        })
+        cls.test_plan = cls.env['mail.activity.plan'].create({
+            'name': 'Test Plan',
+            'company_id': cls.test_company.id,
+            'active': False,
+            'res_model': 'res.partner',
+            'template_ids': [
+                Command.create({
+                    'activity_type_id': cls.activity_type_todo.id,
+                    'responsible_id': cls.user_admin.id,
+                    'responsible_type': 'on_demand',
+                    'sequence': 1,
+                    'summary': 'Test Plan',
+                }),
+            ]
+        })
+
+
+@tagged('mail_activity', 'mail_activity_plan')
+class TestActivityPlan(ActivityPlanScheduleCase):
+
+    def test_plan_copy(self):
+        """Test plan copy"""
+        copied_plan = self.test_plan.copy()
+        self.assertEqual(copied_plan.name, f'{self.test_plan.name} (copy)')
+        self.assertEqual(copied_plan.company_id, self.test_plan.company_id)
+        self.assertEqual(copied_plan.active, self.test_plan.active)
+        self.assertEqual(len(copied_plan.template_ids), len(self.test_plan.template_ids))
+        self.assertEqual(copied_plan.template_ids[0].activity_type_id, self.test_plan.template_ids[0].activity_type_id)
+        self.assertEqual(copied_plan.template_ids[0].responsible_id, self.test_plan.template_ids[0].responsible_id)
+        self.assertEqual(copied_plan.template_ids[0].responsible_type, self.test_plan.template_ids[0].responsible_type)
+        self.assertEqual(copied_plan.template_ids[0].sequence, self.test_plan.template_ids[0].sequence)
+        self.assertEqual(copied_plan.template_ids[0].summary, self.test_plan.template_ids[0].summary)
```

Comment:
Not sure about this as well.. I think `self.assertEqual(len(copied_plan.template_ids), len(self.test_plan.template_ids))` is enough

Reply:
Probably, indeed